docs/user-guide/configure.md

```diff
@@ -254,6 +254,119 @@ But the following pattern would be reported as a warning:
 a[data-auto="1"] {}
 ```
 
+## `languageOptions`
+
+You can customize the syntax to define or extend the syntax for at-rules, properties, types, and CSS-wide keywords.
+
+### `syntax`
+
+You can extend or modify the default CSS syntax to customize the following aspects:
+
+- `atRules`: Define custom at-rules with specific `prelude` and `descriptors` syntax
+- `cssWideKeywords`: Extend the list of CSS-wide keywords with custom values
+- `properties`: Customize the syntax of specific properties
+- `types`: Extend or modify type definitions used in property values
+
+```json
+{
+  "languageOptions": {
+    "syntax": {
```

Comment:
note

As to why `languageOptions` has only one sub-property right now see

> [and tie into a Baseline set in `languageOptions`](https://github.com/stylelint/stylelint/issues/8291#issuecomment-2587132762)

i.e. it won't be the only one